src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -332,13 +334,22 @@ public override ICollection Keys
         {
             get
             {
-                Array arr = new string[base.Keys.Count];
-                base.Keys.CopyTo(arr, 0);
-                Array.Sort(arr, StringComparer.OrdinalIgnoreCase);
-                return arr;
+                ArrayList keyList = new ArrayList(base.Keys);
+                keyList.Sort();
+                MoveItemOnFirstPlace(PSVersionInfo.PSEditionName, keyList);
+                MoveItemOnFirstPlace(PSVersionInfo.PSVersionName, keyList);
+                return keyList.ToArray();
```

Comment:
Thanks for help with the code!
I added the optimized sorting and description. 

If we do `Keys` read-only we should do `PSVersionTable` read-only too. And it is a breaking change - currently we can add in `PSVersionTable` own version strings. For example, third-party module can add its strings in `PSVersionTable`. I think it's not worth and  we can live without the `Keys` cache. What are your thoughts?